internal/resources/free_tier_cluster.go

```diff
@@ -0,0 +1,491 @@
+		return
+	}
+}
+
+func (f *FreeTierCluster) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	// Retrieve values from state.
+	var state providerschema.FreeTierCluster
+	diags := request.State.Get(ctx, &state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resourceIDs, err := state.Validate()
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error deleting free tier cluster",
+			"Could not delete cluster id "+state.Id.String()+" unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	var (
+		organizationId = resourceIDs[providerschema.OrganizationId]
+		projectId      = resourceIDs[providerschema.ProjectId]
+		clusterId      = resourceIDs[providerschema.Id]
+	)
+
+	// Delete existing Cluster.
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/freeTier/%s", f.HostURL, organizationId, projectId, clusterId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodDelete, SuccessStatus: http.StatusAccepted}
+	_, err = f.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		f.Token,
+		nil,
+	)
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.AddError(
+			"Error Deleting Free Tier Cluster",
+			"Could not delete cluster id "+state.Id.String()+": "+errString,
+		)
+		return
+	}
+
+	err = f.checkFreeTierClusterStatus(ctx, state.OrganizationId.ValueString(), state.ProjectId.ValueString(), state.Id.ValueString())
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if !resourceNotFound {
+			response.Diagnostics.AddError(
+				"Error Deleting Capella Cluster",
+				"Could not delete cluster id "+state.Id.String()+": "+errString,
+			)
+			return
+		}
+		// resourceNotFound as expected.
+		return
+	}
+
+	// This case will only occur when cluster deletion has failed,
+	// and the cluster record still exists in the cp metadata. Therefore,
+	// no error will be returned when performing a GET call.
+	cluster, err := f.retrieveFreeTierCluster(ctx, state.OrganizationId.ValueString(), state.ProjectId.ValueString(), state.Id.ValueString())
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.AddError(
+			"Error Deleting Capella Cluster",
+			"Could not delete cluster id "+state.Id.String()+": "+errString,
+		)
+		return
+	}
+	response.Diagnostics.AddError(
+		"Error deleting cluster",
+		fmt.Sprintf("Could not delete cluster id %s, as current Cluster state: %s", state.Id.String(), cluster.CurrentState),
+	)
+}
```

Comment:
I don't think the comment is correct.   if cluster deletion fails, we should return on line 299